lib/pbench/client/__init__.py

```diff
@@ -0,0 +1,289 @@
+            headers: A dictionary of header/value pairs
+            kwargs: Additional `requests` parameters (e.g., params)
+
+        Returns:
+            An HTTP Response object
+        """
+        url = self._uri(api, uri_params)
+        response = self.session.put(
+            url, json=json, headers=self._headers(headers), **kwargs
+        )
+        response.raise_for_status()
+        return response
+
+    def post(
+        self,
+        api: str,
+        uri_params: Optional[JSONOBJECT] = None,
+        *,
+        json: Optional[Dict[str, str]] = None,
+        headers: Optional[Dict[str, str]] = None,
+        **kwargs,
+    ) -> requests.Response:
+        """
+        Issue a post HTTP operation through the cached session, constructing a
+        URI from the "api" name and parameters, adding or overwriting HTTP
+        request headers as specified, and optionally passing a JSON request
+        payload.
+
+        HTTP errors are raised by exception to ensure they're not overlooked.
+
+        Args:
+            api:    The name of the Pbench Server API
+            uri_params: A dictionary of named parameters to expand a URI template
+            json:   A JSON request payload as a Python dictionary
+            headers: A dictionary of header/value pairs
+            kwargs: Additional `requests` parameters (e.g., params)
+
+        Returns:
+            An HTTP Response object
+        """
+        url = self._uri(api, uri_params)
+        response = self.session.post(
+            url, json=json, headers=self._headers(headers), **kwargs
+        )
+        response.raise_for_status()
+        return response
+
+    def delete(
+        self,
+        api: str,
+        uri_params: Optional[JSONOBJECT] = None,
+        *,
+        json: Optional[Dict[str, str]] = None,
+        headers: Optional[Dict[str, str]] = None,
+        **kwargs,
+    ) -> requests.Response:
+        """
+        Issue a delete HTTP operation through the cached session, constructing
+        a URI from the "api" name and parameters, adding or overwriting HTTP
+        request headers as specified.
+
+        HTTP errors are raised by exception to ensure they're not overlooked.
+
+        Args:
+            api:    The name of the Pbench Server API
+            uri_params: A dictionary of named parameters to expand a URI template
+            headers: A dictionary of header/value pairs
+            kwargs: Additional `requests` parameters (e.g., params)
+
+        Returns:
+            An HTTP Response object
+        """
+        url = self._uri(api, uri_params)
+        response = self.session.delete(url, headers=self._headers(headers), **kwargs)
+        response.raise_for_status()
+        return response
+
+    def connect(self) -> None:
+        """
+        Connect to the Pbench Server host using the endpoints API to be sure
+        that it responds, and cache the endpoints response payload.
```

Comment:
We should probably mention here that we also set a default `accept` header for requests made using this session.